<?xml version="1.0" ?>
<launch>
  <arg name="vehicle" default="H01" />
  <arg name="ns" default="/$(arg vehicle)" />
  <!-- Type of node.  Current options are 'robot' and 'base' -->
  <arg name="type" default="robot" />
  <!-- Whether to merge everything (true) or prioritize own map (false) -->
  <arg name="fullMerge" default="false" />
  <!-- Whether to prioritize free space from any agent -->
  <arg name="freePrioritize" default="false" />
  <!-- Whether to publish merged and diff PCL topics -->
  <arg name="publishMergedPCL" default="false" />
  <arg name="publishDiffPCL" default="false" />
  <!-- Octomap type - 0: Binary, 1: Full -->
  <arg name="octoType" default="0" />
  <!-- Map resolution.  If different res needed for each need to change code -->
  <arg name="resolution" default="0.2" />
  <!-- Rate to run the node at -->
  <arg name="rate" default="0.1" />
  <!-- Size of map differences to trigger a merge -->
  <arg name="mapThresh" default="50" />
  <!-- Topics to subscribe and publish -->
  <arg name="mapTopic" default="octomap_binary" />
  <arg name="neighborsTopic" default="neighbor_maps" />
  <arg name="mergedTopic" default="aligned_map" />
  <arg name="mapDiffsTopic" default="map_diffs" />
  <arg name="numDiffsTopic" default="num_diffs" />
  <arg name="pclMergedTopic" default="pc2_merged" />
  <arg name="pclDiffTopic" default="pc2_diff" />

  <node ns="$(arg ns)" name="align_test" pkg="octomap_merger" type="align_test_node" output="screen">
    <param name="vehicle" value="$(arg vehicle)" />
    <param name="type" value="$(arg type)" />
    <param name="fullMerge" value="$(arg fullMerge)" />
    <param name="publishMergedPCL" value="$(arg publishMergedPCL)" />
    <param name="publishDiffPCL" value="$(arg publishDiffPCL)" />
    <param name="octoType" value="$(arg octoType)" />
    <param name="resolution" value="$(arg resolution)" />
    <param name="rate" value="$(arg rate)" />
    <param name="mapThresh" value="$(arg mapThresh)" />
    <param name="mapTopic" value="$(arg mapTopic)" />
    <param name="neighborsTopic" value="$(arg neighborsTopic)" />
    <param name="mergedTopic" value="$(arg mergedTopic)" />
    <param name="mapDiffsTopic" value="$(arg mapDiffsTopic)" />
    <param name="numDiffsTopic" value="$(arg numDiffsTopic)" />
    <param name="pclMergedTopic" value="$(arg pclMergedTopic)" />
    <param name="pclDiffTopic" value="$(arg pclDiffTopic)" />
  </node>
</launch>
